### PR TITLE
Exclude manual segments from dropdown in billboard admin form

### LINKED ISSUE #19578: Two "Managed elsewhere" select options when creating a new Billboard
**Describe the bug**
There are 2 "Managed elsewhere" select options in the "Users who:" dropdown when creating a new billboard.

**To Reproduce**

1. Go to https://dev.to/admin/customization/display_ads/new
2. Select `Only logged in users` for the "Display to user group" select option
3. In the "Users who:" select, you will see that there are two entries (with different numbers set for the `value`) for "Managed elsewhere" 

**Expected behavior**

There should only be one entry in this select dropdown.

**Screenshots**

Screenshot 2023-06-02 at 1 48 45 PM

**Additional context**

Not sure if adding manual audience segment caused this issue.

### PR DESCRIPTION
**Pull request by filleduchaos**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

We had a "bug" where multiple "Managed elsewhere" (the human readable description for manually-managed audience segments) options would appear in the `audience_segment` dropdown when creating a billboard, stemming from the fact that multiple manual segments can exist (unlike the automatically refreshed ones, which are limited to one of each type).

Since we don't have a way to distinguish different manually-managed segments in Forem core, it doesn't quite make sense to have them be a selectable option in the UI at all. This PR removes them from the (default) dropdown list; additionally, if a billboard has been assigned to a manual segment (e.g. through the API) the "Managed elsewhere" text will be displayed but the user will not be able to change that assignment from the admin section.

## Related Tickets & Documents

- Closes #19578 

## QA Instructions, Screenshots, Recordings

https://www.loom.com/share/9f224d3875664ad2940bbbb5d1dca37d

Also noticed after recording that for some reason Loom didn't capture the select dropdown when I opened it in the end, here is a screenshot (showing it only has automatic audience segments now):

Screenshot of display ad admin edit page showing that the 'Users who:' dropdown only shows automatically refreshed segments

### UI accessibility checklist

- [x] Semantic HTML implemented?
- [x] Keyboard operability supported?

## Added/updated tests?

- [x] Yes

## [optional] What gif best describes this PR or how it makes you feel?

Various Spider-people pointing at each other in confusion (from the Across the Spider-Verse trailer)

### CODE REVIEW

**Comment by filleduchaos** [comment]
@benhalpern I thought it would make more sense to leave managing the manual audience segment <-> billboard link entirely to external tools - I'm open to other suggestions though

**Comment by mirie** [comment]
@filleduchaos do you want all the tagged folks to review? Trying to streamline PR reviewer assignments to 2 devs, but open to increasing for this one if you want specific feedback from this set of reviewers.

File: app/helpers/display_ad_helper.rb [hunk 1]
```diff
@@ -3,7 +3,15 @@ def display_ads_placement_area_options_array
...
-  def audience_segments_for_display_ads
-    AudienceSegment.human_readable_segments
+  def automatic_audience_segments_options_array
+    AudienceSegment.not_manual.pluck(:id, :type_of)
+      .map { |(id, type)| [AudienceSegment.human_readable_description_for(type), id] }
+  end
+
+  def single_audience_segment_option(billboard)
+    segment = billboard.audience_segment
+    raise if segment.blank?
```
Comment (lightalloy) [inline app/helpers/display_ad_helper.rb:13]: Let's raise a specific exception?

Comment (filleduchaos) [reply app/helpers/display_ad_helper.rb]: Corrected, thanks! Also added a comment so the expectation is clear

**Pull request merged**